mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -0,0 +1,115 @@
+  }, [blockCount])
+
+  function getPairAmountFromLP (symbol: string, amount: string): { tokenAAmount: BigNumber, tokenBAmount: BigNumber, tokenASymbol: string, tokenBSymbol: string } {
+      const pair = pairs.find(pair => pair.data.symbol === symbol)
+
+      if (pair === undefined) {
+        throw new Error(`The LP symbol is not existing: ${symbol}`)
+      }
+
+      const ratioToTotal = new BigNumber(amount).div(pair.data.totalLiquidity.token)
+      // assume defid will trim the dust values too
+      return {
+        tokenAAmount: ratioToTotal.times(pair.data.tokenA.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN),
+        tokenBAmount: ratioToTotal.times(pair.data.tokenB.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN),
+        tokenASymbol: pair.data.tokenA.symbol,
+        tokenBSymbol: pair.data.tokenB.symbol
+      }
+  }
+
+  function getTokenPrice (symbol: string): BigNumber {
+    // active price for walletTokens based on USDT
+    const selectedPoolPairs = getSelectedPoolPairs(symbol, 'USDT')
+    const { aToBPrice } = calculatePriceRates(symbol, selectedPoolPairs, '1')
+    return aToBPrice
+  }
+
+  function getSelectedPoolPairs (tokenASymbol: string, tokenBSymbol: string): PoolPairData[] {
+    // TODO - Handle cheapest path with N hops, currently this logic finds the shortest path
+    const { path } = findPath(graph, tokenASymbol, tokenBSymbol)
+    return path.reduce((poolPairs: PoolPairData[], token, index): PoolPairData[] => {
+      const pair = pairs.find(pair => checkIfPair({
+        a: pair.data.tokenA.symbol,
+        b: pair.data.tokenB.symbol
+      }, token, path[index + 1]))
+      if ((pair == null) || index === path.length) {
+        return poolPairs
+      }
+      return [...poolPairs, pair.data]
+    }, [])
+  }
+
+  function calculatePriceRates (fromTokenSymbol: string, pairs: PoolPairData[], amount: string): { aToBPrice: BigNumber, bToAPrice: BigNumber, estimated: BigNumber} {
+    let lastTokenBySymbol = fromTokenSymbol
+    let lastAmount = new BigNumber(amount)
+    const priceRates = pairs.reduce((priceRates, pair): { aToBPrice: BigNumber, bToAPrice: BigNumber, estimated: BigNumber } => {
+      const [reserveA, reserveB] = pair.tokenB.symbol === lastTokenBySymbol ? [pair.tokenB.reserve, pair.tokenA.reserve] : [pair.tokenA.reserve, pair.tokenB.reserve]
+      const [tokenASymbol, tokenBSymbol] = pair.tokenB.symbol === lastTokenBySymbol ? [pair.tokenB.symbol, pair.tokenA.symbol] : [pair.tokenA.symbol, pair.tokenB.symbol]
+
+      const priceRateA = new BigNumber(reserveB).div(reserveA)
+      const priceRateB = new BigNumber(reserveA).div(reserveB)
+      // To sequentially convert the token from its last token
+      const aToBPrice = tokenASymbol === lastTokenBySymbol ? priceRateA : priceRateB
+      const bToAPrice = tokenASymbol === lastTokenBySymbol ? priceRateB : priceRateA
+      const estimated = new BigNumber(lastAmount).times(aToBPrice)
+
+      lastAmount = estimated
```

Comment:
what is this line for?

Reply:
Calculating price rates 
a -> b --- b -> c

The `lastAmount` computed from the previous iteration of the loop is needed to estimate the new one. We're sequentially converting from one token to another until we get into the desired token. Alternatively (not tested), 
```
 const estimated = index === 0 ? new BigNumber(amount).times(aToBPrice) : new BigNumber(estimated).times(aToBPrice)
```

to remove
```lastAmount = estimated```


But I prefer storing it in a variable, it is easier to follow 🤔

Reply:
my question is for lastAmount = estimated and the line after, was it used?

Reply:
ok nvm saw it now. It's looping